source/dagger/gameplay/ping_pong/pingpong_playerinput.cpp

```diff
@@ -23,34 +28,107 @@ void PingPongPlayerInputSystem::WindDown()
 
 void PingPongPlayerInputSystem::OnKeyboardEvent(KeyboardEvent kEvent_)
 {
-    Engine::Registry().view<ControllerMapping>().each([&](ControllerMapping& ctrl_)
+    auto view = Engine::Registry().view<ControllerMapping, PlayerPowerUp>();
+
+    for(auto entity : view)
     {
-        if (kEvent_.key == ctrl_.up_key && (kEvent_.action == EDaggerInputState::Pressed || kEvent_.action == EDaggerInputState::Held))
-        {
-            ctrl_.input.y = 1;
-        }
-        else if (kEvent_.key == ctrl_.up_key && kEvent_.action == EDaggerInputState::Released && ctrl_.input.y > 0)
-        {
-            ctrl_.input.y = 0;
-        }
-        else if (kEvent_.key == ctrl_.down_key && (kEvent_.action == EDaggerInputState::Held || kEvent_.action == EDaggerInputState::Pressed))
-        {
-            ctrl_.input.y = -1;
-        }
-        else if (kEvent_.key == ctrl_.down_key && kEvent_.action == EDaggerInputState::Released && ctrl_.input.y < 0)
-        {
-            ctrl_.input.y = 0;
-        }
-    });
+        auto& ctrl_ = view.get<ControllerMapping>(entity);
```

Comment:
we use _ in the end of the name for parameters. for example kEvent_. 
local variables should not have it.

Reply:
Sorry about that , fixed it now